packages/flutter/lib/src/widgets/app.dart

```diff
@@ -1496,7 +1496,16 @@ class _WidgetsAppState extends State<WidgetsApp> with WidgetsBindingObserver {
     if (navigator == null) {
       return false;
     }
-    navigator.pushNamed(route);
+    final Uri uri = routeInformation.uri;
+    navigator.pushNamed(
```

Comment:
Why can we just ignore all other pieces of the URI here? Should we assert that those are empty?